### PR TITLE
[10.0] [FIX] for received invoices date invoice is not relevant, use date instead

### PR DESCRIPTION
**Pull request by sergiocorato**
added more extensive tests splitting on purchase and sale

### CODE REVIEW

File: account_invoice_constraint_chronology/model/account_invoice.py [hunk 1]
```diff
@@ -11,42 +11,36 @@ class AccountInvoice(models.Model):
 
     @api.multi
     def action_move_create(self):
-        previously_validated = self.filtered(lambda inv: inv.move_name)
+        previously_validated = self.filtered(lambda x: x.move_name)
```
Comment (OpenCode) [inline account_invoice_constraint_chronology/model/account_invoice.py:14]: Perché questa modifica? Giusto per curiosità

File: account_invoice_constraint_chronology/model/account_invoice.py [hunk 2]
```diff
@@ -11,42 +11,36 @@ class AccountInvoice(models.Model):
...
+                    else 'date_invoice'
+                date_field_value = inv.date if inv.type in ['in_invoice', 'in_refund'] \
+                    else inv.date_invoice
+                invoices = self.search([
+                    ('state', 'not in', ['open', 'paid', 'cancel', 'proforma',
+                     'proforma2']),
+                    (date_field, '!=', False),
+                    (date_field, '<', date_field_value),
+                    ('journal_id', '=', inv.journal_id.id)
+                ], limit=1)
```
Comment (OpenCode) [inline account_invoice_constraint_chronology/model/account_invoice.py:28]: Non sarebbe il caso di ordinarle per data in modo che date_field_value contenga l'ultima utile?

File: account_invoice_constraint_chronology/model/account_invoice.py (hunk 1 above)
Comment (sergiocorato) [reply account_invoice_constraint_chronology/model/account_invoice.py:14]: English :pray:
Local code validator said variable is used in another scope, infact inv. is used below

File: account_invoice_constraint_chronology/model/account_invoice.py (hunk 2 above)
Comment (sergiocorato) [reply account_invoice_constraint_chronology/model/account_invoice.py:28]: This search is only used to give an error if a record is found, no need to get a particular invoice (`invoices` variable is not used after the search)

**Comment by sergiocorato** [comment]
Close for inactivity

**Pull request closed without merging**